go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -521,6 +521,12 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 		}()
 	}
 
+	ovsClient, err := libovsdb.NewOVSClient(ctx.Done())
+	if err != nil {
+		cancel()
+		return fmt.Errorf("failed to initialize libovsdb vswitchd client: %w", err)
+	}
+
```

Comment:
This should only be created when `runMode.node`

The usage we had below for metrics seems to only be IC compatible.

I would either return an error if metrics have been requested on non-node configuration or alternatively not start the metric server or not register the metrics in that case.

But it seems a somewhat incorrect metrics design. @martinkennelly is there anything I am missing here?

Reply:
whats metrics are you referring to that are ic only?